### PR TITLE
Splitup the Test Macro file, alternative

### PR DESCRIPTION
**Pull request by basvodde**
Alternative split by splitting out String functions

### CODE REVIEW

**Comment by arstrube** [comment]
I have no objections to this solution, but could you please put the new files into sources.mk for Dos and see that the Dos build is indeed fixed. I am not sure about this, as the long long functions are prone to generate a lot of assembler code and I have no idea whether the string funcitons will outweigh them.

**Comment by arstrube** [comment]
Needless to say that TestTestingFixture is cleaner this way :-)

**Comment by basvodde** [comment]
It still needs to be cleaned up a bit though...

**Comment by arstrube** [comment]
I'm kind of fond of my "hack" in the header though. I wouldn't have thought I could make the static functiosn work without a source file :P

**Comment by basvodde** [comment]
I'd prefer moving everything to the source file anyway. It was already something that should have happened.

**Comment by arstrube** [comment]
Can't really dispute that :D.

Are you going to make the changes needed to platforms/dos/sources.mk? Or how should we approach that? Should I submit a PR against basvodde/master?

**Comment by basvodde** [comment]
Yes, will gradually :) Let me see the failure reasons.

**Comment by arstrube** [comment]
For starters and obvious reasons, you need src/CppUTest/TestTestingFixture.cpp in the dos buid....

**Comment by arstrube** [comment]
Same goes for Cmake builds. 

Interesting failures for Autotools, and also interesting that Cygwin succeded :-/

File: src/CppUTest/CMakeLists.txt [hunk 1]
```diff
@@ -14,7 +14,7 @@ set(CppUTest_src
         MemoryLeakDetector.cpp
         TestFilter.cpp
         TestPlugin.cpp
-        TestTestingPlugin.cpp
```
Comment (arstrube) [inline src/CppUTest/CMakeLists.txt]: :-)

Comment (basvodde) [inline src/CppUTest/CMakeLists.txt]: /me an idiot not paying attention :)

**Comment by arstrube** [comment]
The FAIL_LOCATION lines need to be commented with `// LCOV_EXCL_LINE` to make Coveralls happy

**Comment by basvodde** [comment]
I'll merge this, the Vistual Studio files need fixing still though :(

**Pull request merged**